### DISCUSSION TITLE
Is it possible to toggle Environment Variables in Stories?

### DISCUSSION THREAD

**Discussion by giovannivocale-molar** [storybookjs/storybook#24102] [General]
I have a component that toggles the visibility of a div depending if an environment variable is set or not.

Is it possible to toggle the value of the env variable in a story?

```tsx
export function MyComponent() {
  const isVisible = process.env.NEXT_PUBLIC_BUTTON_IS_VISIBLE === "True"

  return (
    <div>
      {isVisible && <button>click</button>}
    </div>
  )
}
```

```tsx
export const ShowButton: Story = {
  env: (config) => ({
    ...config,
    NEXT_PUBLIC_BUTTON_IS_VISIBLE: 'True',
  }),
};
```

**Comment by jonniebigodes** [comment]
@giovannivocale-molar, thanks for opening up this discussion. Appreciate it 🙏 ! Can you check the documentation on how to work with environment variables [here](https://storybook.js.org/docs/react/configure/environment-variables) and see if it helps.

Let us know so that we can continue working with you and try and help with your case.

Have a fantastic day.

Stay safe

**Reply by giovannivocale-molar** [reply]
I had a look but doesn't seem to allow to "mock" a specific environment variable within a component.
